backend/mlarchive/archive/utils.py

```diff
@@ -448,3 +383,197 @@ def move_list(source, target):
+        for member in elist.members.all():
+            MailmanMember.objects.create(email_list=elist, address=member.username)
+
+
+def lookup_user(address):
+    '''
+    This function takes an email address and looks in Datatracker for an associated
+    Datatracker account name.  Returns None if the email is not found or if there is no
+    associated User account.
+    '''
+    # check cache
+    username_map = cache.get('username_map')
+    if username_map and address in username_map:
+        return username_map[address]
+
+    apikey = settings.DATATRACKER_PERSON_ENDPOINT_API_KEY
+    url = settings.DATATRACKER_PERSON_ENDPOINT
+    data = {'apikey': apikey, '_expand': 'user', 'email': address}
+
+    try:
+        response = requests.post(url, data, timeout=settings.DEFAULT_REQUESTS_TIMEOUT)
+    except requests.exceptions.RequestException as error:
+        logger.error(str(error))
+        return None
+
+    if response.status_code != 200:
+        logger.error('Call to %s returned error %s' % (url, response.status_code))
+        return None
+
+    try:
+        output = response.json()
+        person_ids = list(output['person.person'])
+        if not person_ids:
+            logger.warning(f'lookup_user failed for {address}')
+            return None
+        user = output['person.person'][person_ids[0]]['user']
+        if user == 'None':
+            return None
+        username = user['username']
+    except (TypeError, LookupError) as error:
+        logger.error(f'lookup_user json response: {output}')
+        logger.error(str(error))
+        return None
+
+    if username_map is None:
+        username_map = {}
+
+    username_map[address] = username
+    cache.set('username_map', username_map, timeout=None)
+
+    return username
+
+
+def username_to_email(username):
+    apikey = settings.DATATRACKER_PERSON_ENDPOINT_API_KEY
+    url = settings.DATATRACKER_PERSON_ENDPOINT
+    data = {'apikey': apikey, '_expand': 'email_set', 'user__username': username}
+
+    try:
+        response = requests.post(url, data, timeout=settings.DEFAULT_REQUESTS_TIMEOUT)
+    except requests.exceptions.RequestException as error:
+        logger.error(str(error))
+        return None
+
+    if response.status_code != 200:
+        logger.error('Call to %s returned error %s' % (url, response.status_code))
+        return None
+
+    try:
+        output = response.json()
+        person_ids = list(output['person.person'])
+        if not person_ids:
+            logger.warning(f'username_to_email failed for {username}')
+            return None
+        email_set = output["person.person"][person_ids[0]]["email_set"]
+        found_email = next(
+            (email for email, details in email_set.items() if details.get("primary")), None
+        )
+        if not found_email:
+            # try active
+            found_email = next(
+                (email for email, details in email_set.items() if details.get("active")), None
```

Comment:
Is it ok to limit the search to active addresses when doing the username lookup init step? Is it possible it'll cause problems?

Reply:
added inactive emails. commit 8b78cabe91da33bee8e71661eb88b85a9732dc3b